src/core/store/modules/lines/lines.actions.ts

```diff
@@ -130,75 +133,43 @@ const getLines = createAsyncThunk<{ linesData: { [category: string]: SecuredLine
-//     // gets all primary + aux line data avaliable by defeault
-
-//     const selectedLine = LinesSelectors.selectSelectedLinePage(state);
-//     const tokenPrices = TokensSelectors.selectTokenPrices(state);
-
-//     // @TODO check if events exist to
-//     console.log('User Portfolio actions state: ', state);
-//     console.log('User Portfolio actions state: ', state);
-//     console.log('User Portfolio actions tokenPrices: ', tokenPrices);
-
-//     // debugger;
-//     if (selectedLine) {
-//       console.log('User Portfolio actions selectedLine: ', selectedLine);
-//       // console.log('user portfolio')
-//       return { linePageData: selectedLine };
-//     } else {
-//       try {
-//         const linePageData = formatLinePageData(
-//           await creditLineService.getLinePage({ network: state.network.current, id }),
-//           tokenPrices
-//         );
-
-//         console.log('user portfolio getLinePage data ', linePageData);
-
-//         // @TODO dispatch actions to save collateral and events
-//         // enable collateral or add collateral
-//         // save module to Map
-//         // dispatch(CollateralActions.saveModuleToMap(spigot.id, spigot ));
-//         // function does not exist in this file
-//         // save events to map
-//         // dispatch(LineActions.setCreditEvetntsModule(spigot.id, collateralEvents ));
-//         // save events to map
-//         // dispatch(CollateralActions.saveEventsToMap(spigot.id, collateralEvents ));
-
-//         if (!linePageData) throw new Error();
-//         return { linePageData };
-//       } catch (e) {
-//         console.log('failed getting full line page data', e);
-//         return { linePageData: undefined };
-//       }
-//     }
-//   }
-// );
-
 const getLinePage = createAsyncThunk<{ linePageData: SecuredLineWithEvents | undefined }, GetLinePageArgs, ThunkAPI>(
   'lines/getLinePage',
   async ({ id }, { getState, extra, dispatch }) => {
     const state: RootState = getState();
     const { creditLineService } = extra.services;
     // gets all primary + aux line data avaliable by defeault
-    const network = getNetwork(`${state.wallet.networkVersion}`);
-    // const selectedLine = LinesSelectors.selectSelectedLinePage(state);
+
+    const selectedLine = LinesSelectors.selectSelectedLinePage(state);
     const tokenPrices = TokensSelectors.selectTokenPrices(state);
-    const linePageResponse = await creditLineService.getLinePage({
-      network,
-      id,
-    });
+    const network = getNetwork(state.wallet.networkVersion);
+    // const selectedLine = LinesSelectors.selectSelectedLinePage(state);
+    // const linePageResponse = await creditLineService.getLinePage({
+    //   network,
+    //   id,
+    // });
 
-    if (!linePageResponse) {
-      return { linePageData: undefined };
+    // query and add credit and collateral events to pre-existing line
+    console.log('get line selected: ', selectedLine);
+    if (selectedLine) {
```

Comment:
maybe check if the events exist already in selectedLine (should be returned by selector) and if not present then make getLLineEvents request

Reply:
done